docs/topics/gradle.md

```diff
@@ -133,6 +133,158 @@ sourceSets {
 </tab>
 </tabs>
 
+### Check for JVM target compatibility of related compile tasks
+
+In the build module, you may have related compile tasks, for example:
+* `compileKotlin` and `compileJava`
+* `compileTestKotlin` and `compileTestJava`
+
+> `main` and `test` source set compile tasks are not related.
+>
+{type="note"}
+
+For such related tasks, the Kotlin Gradle plugin checks for JVM target compatibility. Different values of `jvmTarget` in the `kotlin` extension
+and [`targetCompatibility`](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)
+in the `java` extension cause incompatibility. For example:
+the `compileKotlin` task has `jvmTarget=1.8`, and
+the `compileJava` task has (or [inherits](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java-extension)) `targetCompatibility=15`.
+
+Control the behavior of this check by setting the `kotlin.jvm.target.validation.mode` property in the `build.gradle` file equal to:
+* `warning` – the default value; the Kotlin Gradle plugin will print a warning message.
+* `error` – the plugin will fail the build.
+* `ignore` – the plugin will skip the check and won't produce any messages.
+
+### Set custom JDK home
+
+> [Setting `jdkHome` option directly is deprecated](https://youtrack.jetbrains.com/issue/KT-46541).
```

Comment:
"The `jdkHome` compiler option is deprecated since Kotlin 1.*.*."

Reply:
Well, here I wouldn't change the wording at least now. By the link, there is the ticket with the target version = 1.7 and the text telling that builds with jdkHome will fail soon. If we change this note to point to the concrete version of Kotlin, the version will be 1.5.30 and we'll lose the link.

Reply:
1. The current wording is not precisely correct - the option itself is deprecated, not "setting it". 
2. This wording doesn't inform about versions in which it was OK to use jdkHome.
3. We describe the current state of things, not our plans. Many times we promised something to happen in the next release and it didn't. This ticket already shows such a case – see discussion in comments and version updates :)